### DISCUSSION TITLE
Trigger a new release as adding a build

### DISCUSSION THREAD

**Discussion by abaganuv** [DarqueWarrior/vsteam#521] [Q&A] [answered]
Hi
When I need to queue a build definition I am using Add-VSTeamBuild -BuildDefinitionId 11111 -BuildParameters with the relevant parameters but when I need to trigger a release definition I cannot pass parameters and need to update the release definition like:
Get-VSTeamReleaseDefinition
Initial parameters ...
$body = $b | ConvertTo-Json -Depth 100 
Update-VSTeamReleaseDefinition -ReleaseDefinition $body 

Why I cannot trigger release as I triggers build?
I have a release that  need to get a path and version  as string and I cannot trigger him

It is possible to add -ReleaseParameters to -Add-VSTeamRelease

**Comment by SebastianSchuetze** [comment, marked as answer]
In Azure DevOps, build pipelines and release pipelines behave differently:

- Build pipelines are designed to accept parameters at queue time because it's common to want to build the same codebase with different parameters. For example, you might want to build a debug version and a release version, or build for different platforms.

- On the other hand, release pipelines are designed around the principle that a given release should be deployable to any environment without modification. This is why you can't pass parameters directly when you trigger a release.

However, there are workarounds to pass variables to a release pipeline:

1. **Using build artifacts**: You can store your parameters in a file during the build, publish that file as a build artifact, and then read the file during the release.

2. **Using Azure DevOps REST API**: You can use the Azure DevOps REST API to manually create a release and set the variables you need. This approach gives you the most control but is also the most complex.

The rest API endpoint can be found here: https://learn.microsoft.com/en-us/rest/api/azure/devops/release/releases/create?view=azure-devops-rest-7.0&tabs=HTTP

**Comment by abaganuv** [comment]
Thank you for your answer
So I will try to use Azure DevOps REST API but I don't have the security (oauth2). when I am using the VSTeam module I am not supplying and security. how can I use the Azure DevOps REST API as I use with the module? I want to not using the security 
How the module is not requiring any security?
Maybe you can send me a block code that can create a release with 2 parameters?  I opened the link you send me and I don't know how to do it, I am talking about PowerShell :) maybe need a function for that
Thanks

**Reply by SebastianSchuetze** [reply]
The module requires Security, because you have to use Set-VSTeamAccount that asks you for a PAT.

But also, I think I didn't express myself clear enough. In short: Classic release pipeline do not support parameters. There are only work arounds. That is why the module can never support parameters either.

Classic pipelines are also not the best choice to use. You should stick dor multi staging pipelines with YAML.
